src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,286 @@
+  labelText: string;
+  /** Optional string used to set the placement of the list element. */
+  listAlignment?: "left" | "right";
+  /** Required array of data objects used to populate the items rendered in the list.
+   * The order of the array elements will determine the order of the items in the list.*/
+  listItemsData: ListItemsData[];
+  /** Optional boolean value used to toggle the visibility of a border around the button element.*/
+  showBorder?: boolean;
+  /** Optional boolean value used to toggle the visibility of the label text for the button element.
+   * If false, this value will be used to set an aria-label attribute on the button element.  */
+  showLabel?: boolean;
+}
+
+/**Type for the icons/images displayed inline with menu items. */
+interface Media {
+  type: string;
+  name?: IconNames;
+  src?: string;
+  alt?: string;
+}
+
+/** The three types of menu items: */
+interface ActionItem {
+  type: "action";
+  label: string;
+  id: string;
+  media?: Media | null;
+  onClick: (id: string) => void;
+}
+
+interface GroupItem {
+  type: "group";
+  id: string;
+  label: string;
+  children: ListItemsData[];
+}
+
+interface DividerItem {
+  type: "divider";
+  id: string;
+}
+
+/** List items can be one of the three types: */
+export type ListItemsData = ActionItem | GroupItem | DividerItem;
+
+const Menu = chakra(
+  forwardRef<HTMLDivElement, MenuProps>(
+    ({
+      className,
+      id,
+      labelText,
+      listAlignment = "left",
+      selectedItem,
+      highlightColor = "blogs",
+      showBorder = true,
+      showLabel = true,
+      listItemsData,
+      ...rest
+    }) => {
+      const styles = useMultiStyleConfig("Menu", {
+        highlightColor,
+        showBorder,
+        showLabel,
+      });
+
+      /**  Handle selection alongside custom onClick behavior. */
+      const [selected, setSelected] = useState(selectedItem);
+      const handleSelect = (id, customHandler) => {
+        if (customHandler) {
+          customHandler(id);
+        }
+        setSelected(id);
+      };
+
+      /**  Check props. */
+      const validateProps = (listItemsData, selectedItem) => {
+        if (!listItemsData) {
+          console.warn(
+            "NYPL Reservoir Menu: The `listItemsData` prop is required."
+          );
+        } else if (
+          selectedItem &&
+          !listItemsData.map((item) => item.id).includes(selectedItem)
+        ) {
+          console.warn(
+            "NYPL Reservoir Menu: The `selectedItem` prop does not match any of the menu items."
+          );
+        } else if (
+          new Set(listItemsData.map((item) => item.id)).size !==
+          listItemsData.length
+        ) {
+          console.warn(
+            "NYPL Reservoir Menu: The `id` values for the list items are not all unique."
+          );
+        } else if (
+          listItemsData.some(
+            (item) =>
+              !item.type ||
+              item.type === "" ||
```

Comment:
I get this typescript issue locally:
![Screen Shot 2024-02-01 at 11 16 52 AM](https://github.com/NYPL/nypl-design-system/assets/1280564/0b872c70-30f4-418d-8784-d56339294cfb)

Reply:
It seems safe to remove because the other two checks are very similar.